### ISSUE TITLE
Typescript 2.9.1 Breaking Changes

### ISSUE THREAD

**Issue by foltz** [ably/ably-js#509]
I tried compiling via Webpack and with typescript 2.9.1 and get the following error:

ERROR in ./node_modules/ably/browser/static/ably-commonjs.js
Module build failed: Error: Debug Failure. False expression.

When I compile using typescript 2.8.4 or earlier, there is no error and the code runs as expected.

The import I'm using looks like:

import * as Ably from 'ably'

I also tried:

import Ably from 'ably'
import * as Ably from 'ably/browser/static/ably-commonjs.js'
import Ably from 'ably/browser/static/ably-commonjs.js'
const Ably = require('ably/browser/static/ably-commonjs.js')

**Comment by funkyboy** [comment]
Thanks for reporting this. I’ll try to reproduce it and let you know.

**Comment by foltz** [comment]
@funkyboy - I can post you a working project for a repro, just let me know.

**Comment by funkyboy** [comment]
@foltz That would definitely help. Thanks :)

**Comment by foltz** [comment]
Hey.  sorry it took me so long to get around to this.  Here's the project.  Details on building/running and repro are in the README

https://github.com/hoperator/ably-webpack-typescript

**Comment by funkyboy** [comment]
@foltz Thanks. I get a different error though :(
```
ably-webpack-typescript> yarn dev
yarn run v1.7.0
$ webpack-dev-server
/Users/cesare/Desktop/ably-webpack-typescript/node_modules/hard-source-webpack-plugin/index.js:104
  );
  ^
SyntaxError: Unexpected token )
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:542:28)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/cesare/Desktop/ably-webpack-typescript/webpack.config.js:5:33)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ably-webpack-typescript>
```

**Comment by funkyboy** [comment]
@foltz Ok, now I can reproduce the bug. I had to `nvm use 8.11.2`
I'll investigate and let you know.

**Comment by mattheworiordan** [comment]
See https://github.com/Microsoft/TypeScript/issues/24577...  Hopefully they can help.

**Comment by funkyboy** [comment]
@foltz Following the trail of issues/PRs:

- https://github.com/Microsoft/TypeScript/issues/24577
- https://github.com/Microsoft/TypeScript/issues/24963
- https://github.com/Microsoft/TypeScript/pull/24973#event-1682298630

It looks like a fix has been merged into the master of TypeScript, but as of now the latest version is still 2.9.2.

I think the next step is to wait for the release of 2.9.3 and check if your project compiles correctly against it.

**Comment by SimonWoolf** [comment]
There haven't been any further releases of typescript 2.9.x, but can confirm upgrading to typescript 3.x fixes this bug (tested with 3.1.2).

**Comment by foltz** [comment]
I've confirmed that everything builds correctly with Typescript 3.x  Guessing you can close #535 as well.

**Comment by SimonWoolf** [comment]
Noting here for the record: while our general advice is to upgrade to typescript 3.x, if for some other reason a customer is stuck on 2.x: while Microsoft has been slow to cut a typescript 2.9.x release that incorporates the fix to this bug, they have merged it into the `release-2.9` branch, so you can reference github + that branch directly to get a 2.9 version that has this fix.
